cpp/dolfinx/fem/DirichletBC.h

```diff
@@ -132,6 +132,71 @@ template <typename T>
 class DirichletBC
 {
 private:
+  /// @brief Build list of cells (local to process, including ghosts)
+  /// which contain any of the input degrees of freedom.
+  ///
+  /// Given a function space `V` and a set of `dofs`, return cell `c` if
+  /// `V.dofmap()->cell_dofs(i)` contains any of the entries in `dofs`.
+  ///
+  /// @param[in] V The function space
+  /// @param[in] dofs A list of degrees of freedom (local indices, unrolled),
+  /// including ghosts
+  /// @returns List of cell indices (local to process, including ghosts) of
+  /// which has any of the input `dofs`
+  std::vector<std::int32_t> locate_cells(const FunctionSpace& V,
```

Comment:
Should this function be static? Or a free function in the `impl` namespace? It's not obvious that it needs to be a non-const member.